src/SerializeTool.cpp

```diff
@@ -0,0 +1,436 @@
+doSerialize(std::string const& data)
+{
+    auto tx = [&]
+    {
+        auto json = serialize::parseJson(data);
+        return json ? serialize::makeObject(json) : boost::none;
+    }();
+    if (!tx)
+    {
+        std::cout << "Unable to serialize \"" << data << "\"" <<
+            std::endl;
+        return;
+    }
+
+    auto result = serialize::serialize(*tx);
+
+    std::cout << result << std::endl;
+}
+
+void
+doDeserialize(std::string const& data)
+{
+    auto result = serialize::deserialize(boost::trim_copy(data));
+
+    if (result)
+        std::cout << result->getJson(0).toStyledString() << std::endl;
+    else
+        std::cout << "Unable to deserialize \"" << data << "\"" <<
+            std::endl;
+}
+
+void
+doSingleSign(std::string const& data,
+    boost::filesystem::path const& keyFile)
+{
+    using namespace serialize;
+    auto tx = make_sttx(boost::trim_copy(data));
+    if (tx)
+    {
+        auto rippleKey = RippleKey::make_RippleKey(keyFile);
+
+        rippleKey.singleSign(*tx);
+
+        std::cout << tx->getJson(0).toStyledString() << std::endl;
+    }
+    else
+        std::cout << "Unable to sign \"" << data << "\"" << std::endl;
+}
+
+void
+doMultiSign(std::string const& data,
+    boost::filesystem::path const& keyFile)
+{
+    using namespace serialize;
+    auto tx = make_sttx(boost::trim_copy(data));
+    if (tx)
+    {
+        auto rippleKey = RippleKey::make_RippleKey(keyFile);
+
+        rippleKey.multiSign(*tx);
+
+        std::cout << tx->getJson(0).toStyledString() << std::endl;
+    }
+    else
+        std::cout << "Unable to sign \"" << data << "\"" << std::endl;
+}
+
+void
+doCreateKeyfile(boost::filesystem::path const& keyFile,
+    std::string const& keytype,
+    boost::optional<std::string> const& seed)
+{
+    using namespace ripple;
+    using namespace serialize;
+
+    if (exists(keyFile))
+        throw std::runtime_error(
+            "Refusing to overwrite existing key file: " +
+            keyFile.string());
+
+    auto kt = [&]
+    {
+        return !keytype.empty() ? keyTypeFromString(keytype) :
+            boost::optional<KeyType>{};
+    }();
+    if (kt && *kt == KeyType::invalid)
+    {
+        std::cout << "Invalid key type: \"" << keytype << "\"" <<
+            std::endl;
+        return;
+    }
+
+    auto key = RippleKey::make_RippleKey(kt, seed);
+
+    key.writeToFile(keyFile);
+
+    std::cout << "New ripple key created.\n" <<
+        "Stored in " << keyFile.string() << ".\n" <<
+        "Key type is " << to_string(key.keyType()) << ".\n" <<
+        "Account ID is " <<
+        toBase58(calcAccountID(key.publicKey())) << ".\n" <<
+        "\n\nThis file should be stored securely and not shared.\n\n";
+
+}
+
+void
+doRepairKeyfile(boost::filesystem::path const& keyFile)
+{
+    using namespace serialize;
+    using namespace boost::filesystem;
+    // Back up the keyfile. If there are already 1000
+    // backups, assume the user knows what they're doing.
```

Comment:
I'm rewriting the interface to `create_keyfile`, and probably getting rid of `repair_keyfile` entirely.